### PR TITLE
PSFR-1827 Add controller unit tests for watchlist routes

### PR DESCRIPTION
**Pull request by abdulrkalam**

### CODE REVIEW

File: server/routes/watchlist/watchlistController.test.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    const postWatchListSpy = jest.spyOn(rpService, 'postWatchlist').mockRejectedValue(new Error('Something went wrong'))
+    await request(app)
+      .post('/addToYourCases?prisonerNumber=123')
+      .expect(500)
+      .expect(res => expect(res.text).toMatchSnapshot())
+
+    expect(postWatchListSpy).toHaveBeenCalledWith('123')
+  })
+  it('Error case - missing parameter', async () => {
+    await request(app).post('/addToYourCases').expect(500)
```
Comment (JamesBoobier) [inline server/routes/watchlist/watchlistController.test.ts:48]: Can you add `.expect(res => expect(res.text).toMatchSnapshot())` to this?

File: server/routes/watchlist/watchlistController.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      .mockRejectedValue(new Error('Something went wrong'))
+    await request(app)
+      .post('/removeFromYourCases?prisonerNumber=123')
+      .expect(500)
+      .expect(res => expect(res.text).toMatchSnapshot())
+
+    expect(deleteWatchListSpy).toHaveBeenCalledWith('123')
+  })
+  it('Error case - missing parameter', async () => {
+    await request(app).post('/removeFromYourCases').expect(500)
```
Comment (JamesBoobier) [inline server/routes/watchlist/watchlistController.test.ts:75]: Same with this one

Comment (abdulrkalam) [reply server/routes/watchlist/watchlistController.test.ts]: done

File: server/routes/watchlist/watchlistController.test.ts (hunk 1 above)
Comment (abdulrkalam) [reply server/routes/watchlist/watchlistController.test.ts]: done

**Review by JamesBoobier — APPROVED** [review]

**Pull request merged**